service-base/src/main/java/org/eclipse/hono/service/VertxBasedHealthCheckServer.java

```diff
@@ -128,17 +118,29 @@ public void setAdditionalResources(final List<Handler<Router>> resourceProviders
         registerAdditionalResources();
 
         final Future<Void> result = Future.future();
-        CompositeFuture.all(bindSecureHttpServer(), bindInsecureHttpServer())
-                .compose(r -> result.complete(), result);
+            bindSecureHttpServer().setHandler(res -> {
+                if (res.failed()){
+                    bindInsecureHttpServer().compose(r -> result.complete(), result);
+                } else if ( (config.isInsecurePortEnabled() && config.getInsecurePort(DEFAULT_PORT) != server.actualPort() )
+                        || (config.getInsecurePort() != Constants.PORT_UNCONFIGURED && config.getInsecurePort() != server.actualPort())) {
+                    bindInsecureHttpServer().compose(r -> result.complete(), result);
+                } else {
+                    result.complete();
+                }
+            });
         return result;
     }
 
     private Future<Void> bindInsecureHttpServer() {
-        if (config.isInsecurePortEnabled()) {
-            final Future<Void> result = Future.future();
+        final Future<Void> result = Future.future();
 
+        if ((config.getInsecurePortBindAddress() == Constants.LOOPBACK_DEVICE_ADDRESS && config.isInsecurePortEnabled())
+             || ( config.getInsecurePortBindAddress() == Constants.LOOPBACK_DEVICE_ADDRESS && config.getInsecurePort() == Constants.PORT_UNCONFIGURED )) {
```

Comment:
IMHO it should be sufficient to check if a bind address has been set here.
The `else` branch should then check if either the `insecurePortEnabled` property has been set to `true` or a port has been set explicitly ...

Reply:
I did ask myself which way to go as well. Both works fine, but checking the port configuration separately allows for a more preciser log message. I'll update in that direction.

Reply:
I updated to match what we agreed on earlier, but I feel there is a little bit of redundancy .